common/g3sinksUtils.hpp

```diff
@@ -0,0 +1,48 @@
+/** ==========================================================================
+* 2020 Joel Stienlet
+* this file is part of g3log by KjellKod.cc 
+*
+* This code is PUBLIC DOMAIN to use at your own risk and comes
+* with no warranties. This code is yours to share, use and modify with no
+* strings attached and no restrictions or obligations.
+* ============================================================================*/
+
+
+
+#ifndef g3DeepCopyTypes
+#define g3DeepCopyTypes
+/*
+ 
+ These deep copy types have been introduced to make it easier to
+ call sink methods with dynamically allocated data, without 
+ having to wait for the worker thread to finish.
+ They work together with the "NoJn"(No thread Join)-suffixed variants of
+ the corresponding sink methods.
+ 
+ Alternatively, you can:
+  - use static data
+  - wait for the thread (join it) with std::future .get()
+
+ */
+#include <functional>
+#include <cstring>
+
+namespace g3
+{
+
+// RAII string type with its deleter (Dltr) included to cross library boundaries 
+class DltrStr
+{
+public:
+    DltrStr() = delete;
+    // the deleter accompanies the memory block to enable it beeing freed in the library (using a different heap, as on Windows):
+    DltrStr(const char * content) {size_t len = strlen(content); _content = std::unique_ptr<char [], std::function<void(char*)>>(new char[len + 1], [](char* b){ delete[](b);} ); memcpy(_content.get(), content, len+1); };
```

Comment:
Please format the code for improved readability. The compiler doesn't care that it's multiple statements on one line or on multiple lines.